core/src/main/java/org/geysermc/geyser/registry/mappings/versions/MappingsReader_v1.java

```diff
@@ -0,0 +1,663 @@
+import org.geysermc.geyser.api.block.custom.component.PlacementConditions.Face;
+import org.geysermc.geyser.api.item.custom.CustomItemData;
+import org.geysermc.geyser.api.item.custom.CustomItemOptions;
+import org.geysermc.geyser.api.util.CreativeCategory;
+import org.geysermc.geyser.item.exception.InvalidCustomMappingsFileException;
+import org.geysermc.geyser.level.block.GeyserCustomBlockComponents.CustomBlockComponentsBuilder;
+import org.geysermc.geyser.level.block.GeyserCustomBlockData.CustomBlockDataBuilder;
+import org.geysermc.geyser.level.block.GeyserMaterialInstance.MaterialInstanceBuilder;
+import org.geysermc.geyser.level.physics.BoundingBox;
+import org.geysermc.geyser.registry.BlockRegistries;
+import org.geysermc.geyser.registry.mappings.util.CustomBlockComponentsMapping;
+import org.geysermc.geyser.registry.mappings.util.CustomBlockMapping;
+import org.geysermc.geyser.registry.mappings.util.CustomBlockStateBuilderMapping;
+import org.geysermc.geyser.registry.mappings.util.CustomBlockStateMapping;
+import org.geysermc.geyser.translator.collision.BlockCollision;
+import org.geysermc.geyser.util.BlockUtils;
+import org.geysermc.geyser.util.MathUtils;
+
+import java.nio.file.Path;
+import java.util.*;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class MappingsReader_v1 extends MappingsReader {
+    @Override
+    public void readItemMappings(Path file, JsonNode mappingsRoot, BiConsumer<String, CustomItemData> consumer) {
+        this.readItemMappingsV1(file, mappingsRoot, consumer);
+    }
+
+    @Override
+    public void readBlockMappings(Path file, JsonNode mappingsRoot, BiConsumer<String, CustomBlockMapping> consumer) {
+        this.readBlockMappingsV1(file, mappingsRoot, consumer);
+    }
+
+    public void readItemMappingsV1(Path file, JsonNode mappingsRoot, BiConsumer<String, CustomItemData> consumer) {
+        JsonNode itemsNode = mappingsRoot.get("items");
+
+        if (itemsNode != null && itemsNode.isObject()) {
+            itemsNode.fields().forEachRemaining(entry -> {
+                if (entry.getValue().isArray()) {
+                    entry.getValue().forEach(data -> {
+                        try {
+                            CustomItemData customItemData = this.readItemMappingEntry(data);
+                            consumer.accept(entry.getKey(), customItemData);
+                        } catch (InvalidCustomMappingsFileException e) {
+                            GeyserImpl.getInstance().getLogger().error("Error in registering items for custom mapping file: " + file.toString(), e);
+                        }
+                    });
+                }
+            });
+        }
+    }
+
+    public void readBlockMappingsV1(Path file, JsonNode mappingsRoot, BiConsumer<String, CustomBlockMapping> consumer) {
```

Comment:
As above, maybe just a basic `@see ...` to the non versioned one?

Reply:
https://github.com/GeyserMC/Geyser/pull/3505/commits/aca66be74a3901d2a8d187e2c7d057e0e7f1b9f5#diff-e4e44067c9d73f54a016c65c7beece03ee0df9ba528e3f044cc371aca53f0f21R79